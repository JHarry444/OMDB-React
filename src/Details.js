import React from 'react';
import axios from 'axios';
import { Image, Container, Row, Col, Table } from 'react-bootstrap';

export default class Details extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            details: {}
        }
    }

    componentDidMount() {
        console.log('props: ', this.props?.match);
        axios.get(`http://www.omdbapi.com/?apikey=335035be&i=${this.props.match?.params?.id || 'fuck'}`)
            .then(response => this.setState({ details: response.data }))
            .catch(err => console.error(err));
    }

    render() {
        return (
            <Container>
                <Row>
                    <Col xs={3}>
                        <Image src={this.state.details.Poster} fluid />
                    </Col>
                    <Col xs={9}  >
                        <Table striped>
                            <tbody>
                                <tr>
                                    <td>Title: </td>
                                    <td>{this.state.details.Title}</td>
                                </tr>
                                <tr>
                                    <td>Year: </td>
                                    <td>{this.state.details.Year}</td>
                                </tr>
                                <tr>
                                    <td>Rated: </td>
                                    <td>{this.state.details.Rated}</td>
                                </tr>
                                <tr>
                                    <td>Actors:</td>
                                    <td> {this.state.details.Actors}</td>
                                </tr>
                                <tr>
                                    <td>Metascore:</td>
                                    <td> {this.state.details.Metascore}</td>
                                </tr>
                                <tr>
                                    <td>Ratings:</td>
                                    <td> {this.state.details.Ratings && this.state.details.Ratings.map(rate => <p>{`${rate.Source}: ${rate.Value}`}</p>)}</td>
                                </tr>
                                <tr>
                                    <td>Plot:</td>
                                    <td> {this.state.details.Plot}</td>
                                </tr>
                            </tbody>
                        </Table>
                    </Col>
                </Row >
            </Container >
        );
    }
}