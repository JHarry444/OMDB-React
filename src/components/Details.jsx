import axios from 'axios';
import { useEffect, useState } from 'react';
import {
  Image, Container, Row, Col, Table,
} from 'react-bootstrap';
import { useParams } from 'react-router-dom';
import { v4 as genId } from 'uuid';

const Details = () => {
  const [details, setDetails] = useState({});
  const { id } = useParams();

  useEffect(() => {
    axios.get(`http://www.omdbapi.com/?apikey=335035be&i=${id}`)
      .then((response) => setDetails(response.data))
      .catch((err) => console.error(err));
  }, []);

  return (
    <Container>
      <Row>
        <Col xs={3}>
          <Image src={details.Poster} fluid />
        </Col>
        <Col xs={9}>
          <Table striped>
            <tbody>
              <tr>
                <td>Title: </td>
                <td>{details.Title}</td>
              </tr>
              <tr>
                <td>Year: </td>
                <td>{details.Year}</td>
              </tr>
              <tr>
                <td>Rated: </td>
                <td>{details.Rated}</td>
              </tr>
              <tr>
                <td>Actors:</td>
                <td>

                  {details.Actors}
                </td>
              </tr>
              <tr>
                <td>Metascore:</td>
                <td>

                  {details.Metascore}
                </td>
              </tr>
              <tr>
                <td>Ratings:</td>
                <td>
                  {details.Ratings && details.Ratings.map((rating) => <p key={genId()}>{`${rating.Source}: ${rating.Value}`}</p>)}
                </td>
              </tr>
              <tr>
                <td>Plot:</td>
                <td>
                  {details.Plot}
                </td>
              </tr>
            </tbody>
          </Table>
        </Col>
      </Row>
    </Container>
  );
};

export default Details;
