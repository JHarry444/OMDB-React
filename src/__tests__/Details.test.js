import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import Details from '../Details';

it('renders the damn component', () => {
  const details = renderer.create(<Details/>);
  expect(details).toMatchSnapshot();
})


it('renders without crashing', () => {
  const details = shallow(<Details />);
  expect(details).toMatchSnapshot();
});

it('renders film correctly', () => {
  const tree = mount(<Details match={{ params: { id: 'tt0389790' } }} />);
  expect(tree).toMatchSnapshot();
});


