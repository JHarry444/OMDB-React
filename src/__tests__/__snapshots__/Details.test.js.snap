// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders film correctly 1`] = `
<Details
  match={
    Object {
      "params": Object {
        "id": "tt0389790",
      },
    }
  }
>
  <Container
    fluid={false}
  >
    <div
      className="container"
    >
      <ForwardRef
        noGutters={false}
      >
        <div
          className="row"
        >
          <Col
            xs={3}
          >
            <div
              className="col-3"
            >
              <Image
                fluid={true}
                rounded={false}
                roundedCircle={false}
                thumbnail={false}
              >
                <img
                  className="img-fluid"
                />
              </Image>
            </div>
          </Col>
          <Col
            xs={9}
          >
            <div
              className="col-9"
            >
              <ForwardRef
                striped={true}
              >
                <table
                  className="table table-striped"
                >
                  <tbody>
                    <tr>
                      <td>
                        Title: 
                      </td>
                      <td />
                    </tr>
                    <tr>
                      <td>
                        Year: 
                      </td>
                      <td />
                    </tr>
                    <tr>
                      <td>
                        Rated: 
                      </td>
                      <td />
                    </tr>
                    <tr>
                      <td>
                        Actors:
                      </td>
                      <td>
                         
                      </td>
                    </tr>
                    <tr>
                      <td>
                        Metascore:
                      </td>
                      <td>
                         
                      </td>
                    </tr>
                    <tr>
                      <td>
                        Ratings:
                      </td>
                      <td>
                         
                      </td>
                    </tr>
                    <tr>
                      <td>
                        Plot:
                      </td>
                      <td>
                         
                      </td>
                    </tr>
                  </tbody>
                </table>
              </ForwardRef>
            </div>
          </Col>
        </div>
      </ForwardRef>
    </div>
  </Container>
</Details>
`;

exports[`renders the damn component 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-3"
    >
      <img
        className="img-fluid"
      />
    </div>
    <div
      className="col-9"
    >
      <table
        className="table table-striped"
      >
        <tbody>
          <tr>
            <td>
              Title: 
            </td>
            <td />
          </tr>
          <tr>
            <td>
              Year: 
            </td>
            <td />
          </tr>
          <tr>
            <td>
              Rated: 
            </td>
            <td />
          </tr>
          <tr>
            <td>
              Actors:
            </td>
            <td>
               
            </td>
          </tr>
          <tr>
            <td>
              Metascore:
            </td>
            <td>
               
            </td>
          </tr>
          <tr>
            <td>
              Ratings:
            </td>
            <td>
               
            </td>
          </tr>
          <tr>
            <td>
              Plot:
            </td>
            <td>
               
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`renders without crashing 1`] = `
<Container
  fluid={false}
>
  <ForwardRef
    noGutters={false}
  >
    <Col
      xs={3}
    >
      <Image
        fluid={true}
        rounded={false}
        roundedCircle={false}
        thumbnail={false}
      />
    </Col>
    <Col
      xs={9}
    >
      <ForwardRef
        striped={true}
      >
        <tbody>
          <tr>
            <td>
              Title: 
            </td>
            <td />
          </tr>
          <tr>
            <td>
              Year: 
            </td>
            <td />
          </tr>
          <tr>
            <td>
              Rated: 
            </td>
            <td />
          </tr>
          <tr>
            <td>
              Actors:
            </td>
            <td>
               
            </td>
          </tr>
          <tr>
            <td>
              Metascore:
            </td>
            <td>
               
            </td>
          </tr>
          <tr>
            <td>
              Ratings:
            </td>
            <td>
               
            </td>
          </tr>
          <tr>
            <td>
              Plot:
            </td>
            <td>
               
            </td>
          </tr>
        </tbody>
      </ForwardRef>
    </Col>
  </ForwardRef>
</Container>
`;
